.github/workflows/windows-build-xpu.yml

```diff
@@ -0,0 +1,156 @@
+name: windows-build-xpu
+
+on:
+  push:
+    tags:
+      - ciflow/xpu/*
+  workflow_dispatch:
+
+      inputs:
+        win-runner:
+            required: false
+            type: string
+            default: "windows.4xlarge.nonephemeral"
+            description: |
+              Label of the runner this job should run on.
+        build-environment:
+            required: false
+            type: string
+            default: "xpu"
+            description: Top-level label for what's being built/tested.
+        s3-bucket:
+            description: S3 bucket to download artifact
+            required: false
+            type: string
+            default: "gha-artifacts" 
+            
+env:
+    conda_env: 'xpu_venv'
+    conda_path: 'C:\Jenkins\Miniconda3'
+    python_version: '3.10'
+    USE_XPU: 1
+    VS2022INSTALLDIR: 'C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools'
+
+jobs:
+  Preparation:
+    runs-on: ["${{ inputs.win-runner }}"]
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
+        with:
+          no-sudo: true
+
+      - name: Install oneAPI bundle
+        shell: cmd
+        run: |
+          if exist "C:\Program Files (x86)\Intel\oneAPI\setvars.bat" (
+            echo "oneAPI bundle already installed"
+          ) else (
+            echo "setvars.bat not found, installing oneAPI bundle"
+            call .ci\pytorch\win-test-helpers\installation-helpers\install_oneapi_bundle.bat
+            echo "oneAPI bundle installed"
+          )
+    
+      - name: Set conda environment
+        shell: cmd
+        run: |
+          call %conda_path%\Scripts\activate.bat
+          call conda remove -n %conda_env% --all -y
+          call conda create -n %conda_env% python=%python_version% -y
+          call conda activate %conda_env%
+          call conda install cmake==3.26.4 ninja -y
+          call conda remove ucrt -y
+          call pip install pyyaml
+          call conda list
+
+  Build-whls:
+    runs-on: ["${{ inputs.win-runner }}"]
+    needs: [Preparation]
+    steps:
+     - name: Build Torch
+       shell: cmd
+       run: |
+         call %conda_path%\Scripts\activate.bat %conda_env%
+         call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"
+         if exist "pytorch" (
+           rmdir /s /q "pytorch"
+         )
+         git clone https://github.com/pytorch/pytorch.git pytorch > git_pytorch_repo_log.txt
+         cd pytorch
+         call conda install conda-forge::rust -y
+         call conda install typing_extensions -y
+         python -m pip install -r requirements.txt > pip_install_pytorch_requirements_log.txt
+         python setup.py bdist_wheel > build_torch_wheel_log.txt
+         
+     - name: Build TorchVision
```

Comment:
Why TorchVision build is part of `ciflow/xpu`?

Reply:
Good point, I have removed build torchvision job from this script and will add it into UT script.